controller/controller.go

```diff
@@ -1319,12 +1321,25 @@ func (ctrl *ProvisionController) updateDeleteStats(volume *v1.PersistentVolume,
 	}
 }
 
-func (ctrl *ProvisionController) updatePersistentVolume(ctx context.Context, volume *v1.PersistentVolume) (*v1.PersistentVolume, error) {
-	newVolume, err := ctrl.client.CoreV1().PersistentVolumes().Update(ctx, volume, metav1.UpdateOptions{})
+// patchPersistentVolumeWithFinalizers patches the PersistentVolume with the given finalizers
+func (ctrl *ProvisionController) patchPersistentVolumeWithFinalizers(ctx context.Context, volume *v1.PersistentVolume, finalizers []string) (*v1.PersistentVolume, error) {
+	oldData, err := json.Marshal(volume)
 	if err != nil {
-		return volume, err
+		return nil, err
+	}
+
+	volumeCopy := volume.DeepCopy()
+	volumeCopy.ObjectMeta.Finalizers = finalizers
+	newData, err := json.Marshal(volumeCopy)
+	if err != nil {
+		return nil, err
+	}
+
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, &v1.PersistentVolume{})
+	if err != nil {
+		return nil, err
 	}
-	return newVolume, nil
+	return ctrl.client.CoreV1().PersistentVolumes().Patch(ctx, volume.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{})
```

Comment:
capture and return results instead of directly making it a part of return. Something like:
```
pv, err := ctrl.client.CoreV1().PersistentVolumes().Patch(ctx, volume.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{})
if err != nil {
 return nil, err
}
return pv
```

Reply:
@deepakkinni done.